web/src/lib/components/NostrChat.svelte

```diff
@@ -0,0 +1,297 @@
+                        <input id="use_browser_extension" type="checkbox" class="toggle toggle-primary mr-2" bind:checked={nostrPreferenceCheckboxChecked} on:change={e => {onChangeNostrPreferenceCheckbox(e)}} disabled />
+                        <span class="label-text">Use browser extension's identity</span>
+                    </label>
+                </div>
+                {#if nostrPreferenceCheckboxChecked}
+                    <small>You'll be asked to authorize signing messages when you want to write in the channel.</small>
+                {:else}
+                    <small>You'll be using your Plebeian Market generated Nostr identity. Install a Nostr browser extension
+                        (<a class="link" href="https://github.com/fiatjaf/nos2x" target="_blank">nos2x</a>,
+                        <a class="link" href="https://getalby.com/" target="_blank">Alby</a> or
+                        <a class="link" href="https://www.blockcore.net/wallet" target="_blank">Blockcore</a>) if you want to use your own Nostr identity:</small>
+                {/if}
+            </div>
+        </div>
+
+        <div class="flex border-base-300 bg-info-content-200 rounded-b-box rounded-tr-box
+                        min-h-[6rem] min-w-[18rem] max-w-4xl flex-wrap items-center justify-center
+                        gap-2 overflow-x-hidden border bg-cover bg-top p-4"
+             style="background-size: 5px 5px; background-image: radial-gradient(hsla(var(--bc)/.2) 0.5px,hsla(var(--b2)/1) 0.5px);"
+        >
+            <div class="w-full">
+                {#each sortedMessages as m}
+                    <div class="chat chat-start" class:mt-0={m.samePubKey === true} class:mt-5={m.samePubKey !== true} class:profileInfo={m.samePubKey !== true}>
+                        <div class="chat-image avatar">
+                            <div class="w-12 rounded-full ring-primary ring-offset-base-100 ring-offset-2" class:ring={m.samePubKey !== true}>
+                                {#if m.samePubKey !== true}
+                                    <img src="{m.profileImage}" alt="profile picture" class:profileInfoImage={m.samePubKey !== true} />
+                                {/if}
+                            </div>
+                        </div>
+                        {#if m.samePubKey !== true}
+                            <div class="chat-header">
+                                <span class="mr-3" class:profileInfoName={m.samePubKey !== true}>{m.profileName}</span>
+                            </div>
+                        {/if}
+                        <div class="chat-bubble">{@html m.content}</div>
+                        <div class="chat-footer text-xs opacity-50">{formatTimestamp(m.created_at)}</div>
+                    </div>
+                {/each}
+            </div>
+        </div>
+    </div>
+
+    {#if emptyChatShowsLoading && sortedMessages.length === 0}
+        <Loading />
```

Comment:
If it's enabled, yes. For stalls it's disabled because most will be empty and we don't want to keep the loading on the screen. For the Market Square, I have it enabled because if there is no message, then we haven't yet connected successfully to any good relay.

But I think we should probably use a less-intrusive loading thing? There is a planed feature to show how many relays are you connected to. Maybe we could show there a spinning wheel until at least x relays connected or one message received...

Reply:
Yep, sounds good. Indeed, I think the "loading" indicator should eventually be based on what is actually happening (connecting to relays) rather than based on the # of messages...